### ISSUE TITLE
sqlite3.DatabaseError: malformed database schema (redactions_have_censored_ts)

### ISSUE THREAD

**Issue by Linuxine** [matrix-org/synapse#6673]
Hi,

I have a running self hosted Matrix homeserver for quite a few years now, and everything worked really fine, but it seems that migration from 1.7.3 to 1.8 broke my instance. I cannot manage to launch matrix (I am using the following command : 

```
/home/matrix/synapse/bin/python -B -m synapse.app.homeserver -c /home/matrix/synapse/homeserver.yaml
```

and I get the following error:

```
2020-01-09 17:46:04,924 - twisted - 192 - ERROR - -     db_conn, engine, hs.config, data_stores=database_config.data_stores,
2020-01-09 17:46:04,924 - twisted - 192 - ERROR - -   File "/home/matrix/synapse/lib/python3.6/site-packages/synapse/storage/prepare_database.py", line 64, in prepare_database
2020-01-09 17:46:04,924 - twisted - 192 - ERROR - -     version_info = _get_or_create_schema_state(cur, database_engine)
2020-01-09 17:46:04,925 - twisted - 192 - ERROR - -   File "/home/matrix/synapse/lib/python3.6/site-packages/synapse/storage/prepare_database.py", line 530, in _get_or_create_schema_state
2020-01-09 17:46:04,925 - twisted - 192 - ERROR - -     executescript(txn, schema_path)
2020-01-09 17:46:04,925 - twisted - 192 - ERROR - -   File "/home/matrix/synapse/lib/python3.6/site-packages/synapse/storage/prepare_database.py", line 524, in executescript
2020-01-09 17:46:04,926 - twisted - 192 - ERROR - -     txn.execute(statement)
2020-01-09 17:46:04,926 - twisted - 192 - ERROR - - sqlite3.DatabaseError: malformed database schema (redactions_have_censored_ts) - near "WHERE": syntax error
```

What should I do ? Is there a workaround ? Thanks a lot for the help !

**Comment by Linuxine** [comment]
My bad, sqlite configuration error, all good now ! :D

**Comment by richvdh** [comment]
?

This sounds like a duplicate of #6426, but that was fixed in synapse 1.7.0

**Comment by rugalash** [comment]
@Linuxine can you actually post what you did that fixed it instead of "I fixed it thx" and nobody knows what went on ever again?

**Comment by Linuxine** [comment]
Hi @rugalash, actually, I was getting this error because I was trying to launch Matrix manually on my CentOS 7, but I had forgotten that the sqlite version packaged in CentOS 7 is too old for Matrix. I did not have the problem when launching matrix as a service, because I added the path to a newer sqlite I compiled in the service file. So to be able to fix my problem, all I had to do was to use the right version of sqlite (> 3.8 if I am not mistaken), and this error message was gone. Hope it helps !
